vendor/visualmetrics.py

```diff
@@ -1457,6 +1462,41 @@ def calculate_speed_index(progress):
         last_progress = p['progress'] / 100.0
     return int(si)
 
+def calculate_contentful_speed_index(progress, directory):
+    x = len(progress)
+    dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), directory)
+    first_paint_frame = os.path.join(
+        dir, "ms_{0:06d}.png".format(
+            progress[1]["time"]))
+    target_frame = os.path.join(
+        dir, "ms_{0:06d}.png".format(progress[x - 1]["time"]))
+    logging.debug("Target image for contSI is %s" % target_frame)
+    content = []
+    maxContent=0
+    for p in progress[1:]:
+        # Full Path of the Current Frame
+        current_frame = os.path.join(dir, "ms_{0:06d}.png".format(p["time"]))
+        logging.debug("Current Image is %s" % current_frame)
+        # Takes full path of PNG frames to compute contentfulness value
+        command = '{0} {1} -canny 2x2+8%+8% -define histogram:unique-colors=true -format %c histogram:info:-'.format(
+            image_magick['convert'], current_frame)
+        output = subprocess.check_output(command, shell=True)
+        value  = int(output.split()[7].split(':')[0])
+        if value > maxContent:
+            maxContent = value
+        content.append(value)
+
+    for i,value in enumerate(content):
+        content[i] = float(content[i]) / float(maxContent)
+
+    # Assume 0 content for first frame
+    cont_si = 1 * (progress[1]['time']-progress[0]['time'])
```

Comment:
nit: it's usual to put spaces between operators, here, in the `xrange` below, in the `i, value` in enumeration loop above, but I just don't care that much for this code.  Up to you.